src/stanchion_stats.erl

```diff
@@ -0,0 +1,205 @@
+
+         {[bucket, create, time], histogram, [],
+          [{mean  , bucket_create_time_mean},
+           {median, bucket_create_time_median},
+           {95    , bucket_create_time_95},
+           {99    , bucket_create_time_99},
+           {100   , bucket_create_time_100}]},
+         {[bucket, delete, time], histogram, [],
+          [{mean  , bucket_delete_time_mean},
+           {median, bucket_delete_time_median},
+           {95    , bucket_delete_time_95},
+           {99    , bucket_delete_time_99},
+           {100   , bucket_delete_time_100}]},
+         {[bucket, put_acl, time], histogram, [],
+          [{mean  , bucket_put_acl_time_mean},
+           {median, bucket_put_acl_time_median},
+           {95    , bucket_put_acl_time_95},
+           {99    , bucket_put_acl_time_99},
+           {100   , bucket_put_acl_time_100}]},
+
+         {[user, create], spiral, [],
+          [{one, user_create}, {count, user_create_total}]},
+         {[user, update], spiral, [],
+          [{one, user_update}, {count, user_update_total}]},
+
+         {[user, create, time], histogram, [],
+          [{mean  , user_create_time_mean},
+           {median, user_create_time_median},
+           {95    , user_create_time_95},
+           {99    , user_create_time_99},
+           {100   , user_create_time_100}]},
+         {[user, update, time], histogram, [],
+          [{mean  , user_update_time_mean},
+           {median, user_update_time_median},
+           {95    , user_update_time_95},
+           {99    , user_update_time_99},
+           {100   , user_update_time_100}]},
+         
+         {[waiting, time], histogram, [],
+          [{mean  , waiting_time_mean},
+           {median, waiting_time_median},
+           {95    , waiting_time_95},
+           {99    , waiting_time_99},
+           {100   , waiting_time_100}]}
+        ]).
+
+%% ====================================================================
+%% API
+%% ====================================================================
+
+
+
+-spec safe_update(metric_name(), integer()) -> ok | {error, any()}.
+safe_update(BaseId, ElapsedUs) ->
+    %% Just in case those metrics happen to be not registered; should
+    %% be a bug and also should not interrupt handling requests by
+    %% crashing.
+    try
+        update(BaseId, ElapsedUs)
+    catch T:E ->
+            lager:error("Failed on storing some metrics: ~p,~p", [T,E])
+    end.
+
+-spec update(metric_name(), integer()) -> ok | {error, any()}.
+update(BaseId, ElapsedUs) ->
+    ok = exometer:update([stanchion|BaseId], 1),
+    ok = exometer:update([stanchion|BaseId]++[time], ElapsedUs).
+
+update(BaseId, ServiceTime, WaitingTime) ->
+    update(BaseId, ServiceTime),
+    ok = exometer:update([stanchion, waiting, time], WaitingTime).
+
+-spec update_with_start(metric_name(), erlang:timestamp()) ->
+                                   ok | {error, any()}.
+update_with_start(BaseId, StartTime) ->
```

Comment:
Could be used later when we add metrics on latencies. Could you defer this comment next round of review after I add metrics on PB API-related metrics?